crates/keycloak/src/validation/updater.rs

```diff
@@ -260,6 +296,232 @@ async fn update_realm_settings(
+    ctx.keycloak().get_flow_executions(realm, flow_alias).await
+}
+
+async fn make_executions_required(
+    ctx: &Ctx<'_>,
+    realm: &str,
+    executions: Vec<AuthenticationExecutionInfoRepresentation>,
+) -> anyhow::Result<()> {
+    let flow_alias = "browser_email_otp";
+    let mut browser_email_otp_form_execution = executions
+        .iter()
+        .find(|&execution| execution.display_name == Some("browser_email_otp forms".to_string()))
+        .unwrap()
+        .clone();
+    browser_email_otp_form_execution.requirement = Some("REQUIRED".to_string());
+
+    ctx.keycloak()
+        .modify_flow_execution(realm, flow_alias, browser_email_otp_form_execution)
+        .await?;
+
+    let mut email_2fa_execution = executions
+        .iter()
+        .find(|&execution| execution.display_name == Some("Email_2FA".to_string()))
+        .unwrap()
+        .clone();
+    email_2fa_execution.requirement = Some("REQUIRED".to_string());
+
+    let mut email_totp_auth_execution: AuthenticationExecutionInfoRepresentation = executions
+        .iter()
+        .find(|&execution| execution.display_name == Some("Email TOTP Authentication".to_string()))
+        .unwrap()
+        .clone();
+
+    email_totp_auth_execution.requirement = Some("REQUIRED".to_string());
+
+    ctx.keycloak()
+        .modify_flow_execution(realm, flow_alias, email_2fa_execution)
+        .await?;
+    ctx.keycloak()
+        .modify_flow_execution(realm, flow_alias, email_totp_auth_execution)
+        .await?;
+
+    Ok(())
+}
+
+async fn create_email_otp_authenticator_execution(
+    ctx: &Ctx<'_>,
+    realm: &str,
+) -> anyhow::Result<()> {
+    let mut body: HashMap<String, Value> = HashMap::new();
+    body.insert(
+        "provider".to_string(),
+        serde_json::Value::String("emailotp-authenticator".to_string()),
+    );
+    ctx.keycloak()
+        .create_flow_execution(realm, "Email_2FA", body)
+        .await?;
+
+    Ok(())
+}
+
+async fn add_configuration_to_browser_email_totp_execution(
+    ctx: &Ctx<'_>,
+    realm: &str,
+    executions: Vec<AuthenticationExecutionInfoRepresentation>,
+) -> anyhow::Result<()> {
+    let mut config: HashMap<String, String> = HashMap::new();
+    config.insert("default.reference.value".to_string(), "".to_string());
+    config.insert("default.reference.maxAge".to_string(), "".to_string());
+    config.insert("simulation".to_string(), "false".to_string());
+    config.insert(
+        "emailSubject".to_string(),
+        ctx.cfg()
+            .keycloak()
+            .authenticator_email_subject()
+            .unwrap_or("Temporary Authentication Code")
+            .to_string(),
+    );
+    config.insert("length".to_string(), "6".to_string());
+    config.insert("ttl".to_string(), "300".to_string());
+    config.insert("maxRetries".to_string(), "3".to_string());
+    config.insert("allowUppercase".to_string(), "true".to_string());
+    config.insert("true".to_string(), "true".to_string());
```

Comment:
These two lines look a little bit suspicious. The second definitely not needed, as it is map, but I doubt about first one as well.